### DISCUSSION TITLE
`darkMode: "selector"` not working from with Tailwind CLI 3.4.1

### DISCUSSION THREAD

**Discussion by thepra** [tailwindlabs/tailwindcss#12996] [Help] [answered]
**What version of Tailwind CSS are you using?**

v3.4.1

**What build tool (or framework if it abstracts the build tool) are you using?**

`npx tailwindcss <arguments...> --watch`

**What version of Node.js are you using?**

v21.6.2

**What browser are you using?**

Edge

**What operating system are you using?**

Windows 10

**Reproduction URL**

https://play.tailwindcss.com/rGMjIJqkQB?size=540x720

**Describe your issue**

in the components `@layer` I define my button, both for the default "light" mode and also `dark:` mode, but as you see, after putting setting up `<html class="dark">...` nothing changes, the `dark:` styling isn't applying, meanwhile if I put the `dark` class at the element level, only then it's working but the codebase is already big and going around and implementing putting such class everywhere is not feasible. 

Is this a known issue of v3.4.1 or am I doing something wrong?

**Comment by adamwathan** [comment]
Hey! This does look like it's working to me, the trick in Play at least is you can't use the `html` element because there's already an `html` element in the iframe being rendered, so that's why it wasn't appearing to work in your demo:

https://play.tailwindcss.com/okPHo69eKQ?size=540x720

But should totally be able to put it on the `html` element in a real project. Do you have a reproduction you can share as a GitHub repo?

**Reply by thepra** [reply]
That's the Play limitation(*), from what I can see, all the `dark:` prefixes in the `@layer`s are skipped through the running watcher service outputting all the other styles meanwhile missing the `dark:` :/

(*) by the way, how can you play if you can't change fondamental pieces of code of that playground?

**Reply by doojinkang** [reply]
In tailwind.config
darkMode: ['selector', 'div.outer[class~="dark"]']

```
<div class="outer dark">
  <button type="button" class="btn">ciao</button>
</div>
```

**Reply by thepra** [reply, marked as answer]
I'm sorry, just the `selector` part works as expected by checking out the output of the compilation:
![immagine](https://github.com/tailwindlabs/tailwindcss/assets/7032509/82d6a12c-5537-45a5-927d-4e2a01974f9e)

but after that I have `gulp watch` that's looking for files updates and if there are it follows scss compilation, bundling and minification, but at the minification step it does seem that the `gulp-clean-css` isn't working correctly with the `:where` CSS parts and discards them completely from the final `style.min.css`, so after commenting the cleaning in pipe it does work as expected:

![immagine](https://github.com/tailwindlabs/tailwindcss/assets/7032509/e09b5656-a06d-49b0-af9c-56d9d6ab9e2d)

Thanks for the support and I hope this might help others that might face a similar issue with gulp 😅
